src/sentry/web/frontend/auth_login.py

```diff
@@ -668,6 +668,22 @@ def handle_basic_auth(self, request: Request, **kwargs) -> HttpResponseBase:
                 if not user.is_active:
                     return self.redirect(reverse("sentry-reactivate-account"))
                 if organization:
+                    # Check if the user is a member of the provided organization based on their email
+                    membership = organization_service.check_membership_by_email(
+                        email=user.email, organization_id=organization.id
+                    )
+
+                    # If the user is a member, the user_id is None, and they are in a "pending invite acceptance" state with a valid invitation link,
+                    # we redirect them to the invitation page to explicitly accept the invite
+                    if (
+                        membership
+                        and membership.user_id is None
+                        and membership.is_pending
+                        and membership.invitation_link
+                    ):
+                        accept_link_position = membership.invitation_link.find("/accept")
+                        return self.redirect(membership.invitation_link[accept_link_position:])
```

Comment:
Why do you need to slice the URL?

Reply:
It doesn't work with the domain. For example: `http://testserver/accept/13/2eef892424458f4b071e581e55d31698/` doesn't work."

Reply:
maybe because it has to contain the region? is there a function I can use?

Reply:
Instead of string slicing, we could use [urlparse](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse) and extract the path + query from that structure.

Reply:
I applied this feedback https://github.com/getsentry/sentry/pull/82005#discussion_r1918231952. would that work?